### ISSUE TITLE
Warn on embedded new lines in CSV/TSV import

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#724]
Although cells can have multi-line values, it's unusual.  It would probably be helpful to users to warn when we encounter embedded newlines in cell values on import.

Related email thread: https://groups.google.com/g/openrefine/c/hmnjFVSINaY/m/sfdXqpja8lcJ

**Comment by wetneb** [comment]
That would be very useful, I have run into this problem multiple times.

**Comment by wetneb** [comment]
The main question that needs to be resolved for this is how can an importer generate warnings and how should they be displayed in the UI?

This particular warning is likely to go undetected at the preview stage, because an embedded newline could well occur after the first 100 lines of the file. So that means we need to wait for project creation, but then it is kind of too late…

**Comment by thadguidry** [comment]
I see this fitting better immediately after project creation.  Informing the user somewhere along the top area where we inform to download latest OpenRefine or show processing of reconciling.  It should stay until the user clicks their choice.  The operation is stored as a Change and allows to Undo/Redo.

_**Warning:**  Your data contains New Lines embedded in cells.  Do you want to "create new rows" for these lines ? Yes/No/Cancel_

(Informatica Powercenter has an optional detector for embedded newlines that can throw an ERROR and pauses ETL operations if so desired.)

**Comment by wetneb** [comment]
It would be useful to have an example dataset which should trigger such a warning. In my experience, this happens when the dataset contains unescaped quote symbols, forcing the parser to parse multiple actual lines into a single cell.
Here is a synthetic example:
```
a,b
c,"d,
e,f
g,h
```
(it would still be better to have a real-world one to confirm)

**Comment by tfmorris** [comment]
I've updated the description with a link to the email thread that prompted this. The original problem was triggered by a bug in OpenRefine's quote processing, but, as @wetneb says, it also could be triggered by a bugging CSV writer (of which there are more than a few). Probably the most common way would be home grown variants where people just do `",".join(fields)` or `'"'+'","'.join(fields)+'"'` with arbitrary text without quoting/escaping anything. The latter example with an unescaped `fields` value array of `["c",""d"]` would produce the example above.

**Comment by thadguidry** [comment]
@wetneb Yes, there's an example in the RFC 4180 (where it mentions "should" and is not strict...it's a misnomer, and I've never seen this supposedly "strict" memo that people sometimes mention, ever.)
 
```
   6.  Fields containing line breaks (CRLF), double quotes, and commas
       should be enclosed in double-quotes.  For example:

       "aaa","b CRLF
       bb","ccc" CRLF
       zzz,yyy,xxx
```

+1 agree that we warn users.  Give them options to deal with it later after import...as part of a cleaning task.  I've always that we should be giving users more hints by having Rules...and even making OpenRefine pluggable for that as part of the whole Data Validation issues we have.

**Comment by tfmorris** [comment]
That example is valid (quoted) CSV and should parse fine. As an aside, there are a couple of [errata](https://www.rfc-editor.org/errata_search.php?rfc=4180) against that example, including one that points out that if you don't make quoting embedded commas mandatory, it makes the file pretty hard (ie impossible) to parse.

**Comment by thadguidry** [comment]
@tfmorris Ah thanks, yes, you seem to have found that mystical "strict" **memo**...which is actually **errata** against the original RFC4 1080 memo.  Thanks so much for finding that for us!
